File: updater/src/main/java/com/legatotechnologies/updater/IForceUpdate.java
package com.legatotechnologies.updater;

import android.view.View;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.StyleRes;

import org.json.JSONObject;

/**
 * Created by davidng on 6/21/17.
 */

interface IForceUpdate {
    ForceUpdate setNotificationTime(long number, int type);

    ForceUpdate setNotificationTime(long millisecond);

    ForceUpdate setJSON(JSONObject jsonObject);

    ForceUpdate setJSONString(String jsonString);

    ForceUpdate setTheme(@StyleRes int style);

    ForceUpdate setLang(Language lang);

    ForceUpdate setCustomView(@NonNull View view);

    ForceUpdate setCustomView(@LayoutRes int resId);

    ForceUpdate setShouldHideButtons(boolean disabled);

    ForceUpdate setShouldHideTitle(boolean disabled);

    ForceUpdate setOptionalListener(UtilsDialog.OnOptionalDialogDismissListener listener);

    ForceUpdate start();
}

File: updater/src/main/java/com/legatotechnologies/v2/updater/IForceUpdate.kt
package com.legatotechnologies.v2.updater

import android.view.View
import androidx.annotation.DimenRes
import androidx.annotation.LayoutRes
import androidx.annotation.StyleRes
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.LifecycleOwner
import com.legatotechnologies.v2.updater.datas.enums.UpdateType
import java.util.*

interface IForceUpdate {
    fun setLifecycle(owner: LifecycleOwner): IForceUpdate
    fun setNotificationTime(millisecond: Long): IForceUpdate
    fun setUrl(url: String): IForceUpdate
    fun setVersion(version: String): IForceUpdate
    fun setContent(content: String): IForceUpdate
    fun setRounded(@DimenRes radius: Int): IForceUpdate
    fun setForceUpdate(isForceUpdate: Boolean): IForceUpdate
    fun setLocale(locale: Locale): IForceUpdate
    fun setTheme(@StyleRes style: Int): IForceUpdate
    fun setCustomView(view: View): IForceUpdate
    fun setCustomView(@LayoutRes resId: Int): IForceUpdate
    fun setOptionalListener(listener: OnOptionalDialogDismissListener?): IForceUpdate
    fun start(manager: FragmentManager)
}

fun interface OnOptionalDialogDismissListener {
    fun onOptionalDialogDismiss()
}